### PR TITLE
Update ActivateAchievementWorker

### PR DESCRIPTION
**Pull request by MadSpyxFR**
Add new error message if achievment name is not valid

### CODE REVIEW

File: src/greenworks_async_workers.cc [hunk 1]
```diff
@@ -142,9 +142,16 @@ ActivateAchievementWorker::ActivateAchievementWorker(
...
-  steam_user_stats->SetAchievement(achievement_.c_str());
-  if (!steam_user_stats->StoreStats())
+  if (!steam_user_stats->SetAchievement(achievement_.c_str())) {
+    SetErrorMessage("Achievement name is not valid.");
+  } else if (!steam_user_stats->StoreStats()) {
     SetErrorMessage("Error on storing user achievement");
+  }
+}
+  
+void ActivateAchievementWorker::HandleOKCallback() {
```
Comment (hokein) [inline src/greenworks_async_workers.cc]: You don't need override this method as the default implementation provided by `Nan::AsyncWorker` is sufficient here.

File: src/greenworks_async_workers.cc [hunk 2]
```diff
@@ -142,9 +142,16 @@ ActivateAchievementWorker::ActivateAchievementWorker(
 void ActivateAchievementWorker::Execute() {
   ISteamUserStats* steam_user_stats = SteamUserStats();
 
-  steam_user_stats->SetAchievement(achievement_.c_str());
-  if (!steam_user_stats->StoreStats())
+  if (!steam_user_stats->SetAchievement(achievement_.c_str())) {
+    SetErrorMessage("Achievement name is not valid.");
+  } else if (!steam_user_stats->StoreStats()) {
```
Comment (hokein) [inline src/greenworks_async_workers.cc]: The new code will change the behavior. We need to call `StoreStats` after `SetAchievement`. So the code should be corrected like:

```
if (!steam_user_stats->SetAchievement(achievement_.c_str())) {
  SetErrorMessage("Achievement name is not valid.");
  return;
}
if (!steam_user_stats->StoreStats()) {
  // ...
}
```

Comment (MadSpyxFR) [inline src/greenworks_async_workers.cc]: So I suggest to change the code to something like the ClearAchievementWorker method:

`  if (!success_) {
    SetErrorMessage("Achievement name is not valid.");
  } else if (!steam_user_stats->StoreStats()) {
    SetErrorMessage("Fails on uploading user stats to server.");
  }
}

void ClearAchievementWorker::HandleOKCallback() {
  Nan::HandleScope scope;
  callback->Call(0, NULL);
}`

**Comment by MadSpyxFR** [comment]
And in greenworks_api.cc 

```
  if (success) {
    ISteamUserStats* stream_user_stats = SteamUserStats();
    stream_user_stats->RequestCurrentStats();
  }
```
steam_user_stats ? or stream_user_stat? even if it doesn't matter

**Comment by hokein** [comment]
Yeah, it's a typo. I'm +1 on changing "stream_user_stats" to  "steam_user_stats" .

**Comment by MadSpyxFR** [comment]
Typo fixed in greenworks_api.cc
Change done in steam_worker

**Comment by hokein** [comment]
Looks good, thanks.

**Pull request merged**